src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/SvDiscoverFromLocalAssemblyContigAlignmentsSpark.java

```diff
@@ -272,44 +281,197 @@ public static AssemblyContigsClassifiedByAlignmentSignatures preprocess(final Sv
+                toolLogger);
 
         final JavaRDD<VariantContext> complexVariantsRDD = ctx.parallelize(complexVariants);
         final SegmentedCpxVariantSimpleVariantExtractor.ExtractedSimpleVariants reInterpretedSimple =
                 SegmentedCpxVariantSimpleVariantExtractor.extract(complexVariantsRDD, svDiscoveryInputMetaData, assemblyRawAlignments);
         final SAMSequenceDictionary refSeqDict = svDiscoveryInputMetaData.getReferenceData().getReferenceSequenceDictionaryBroadcast().getValue();
-        final Logger toolLogger = svDiscoveryInputMetaData.getToolLogger();
         final String derivedOneSegmentSimpleVCF = outputPrefixWithSampleName + "cpx_reinterpreted_simple_1_seg.vcf";
         final String derivedMultiSegmentSimpleVCF = outputPrefixWithSampleName + "cpx_reinterpreted_simple_multi_seg.vcf";
         SVVCFWriter.writeVCF(reInterpretedSimple.getReInterpretZeroOrOneSegmentCalls(), derivedOneSegmentSimpleVCF, refSeqDict, toolLogger);
         SVVCFWriter.writeVCF(reInterpretedSimple.getReInterpretMultiSegmentsCalls(), derivedMultiSegmentSimpleVCF, refSeqDict, toolLogger);
+
+        return new CpxAndReInterpretedSimpleVariants(complexVariants, reInterpretedSimple.getMergedReInterpretCalls());
+    }
+
+    // TODO: 6/29/18 when BND variants are interpreted using short read evidence (e.g. EvidenceTargetLinks, resolved inversions), put it here
+    private static List<VariantContext> extractInversions() {
+        return Collections.emptyList();
+    }
+
+    //==================================================================================================================
+
+    public static void filterAndWriteMergedVCF(final String outputPrefixWithSampleName,
+                                               final List<VariantContext> variants,
+                                               final SvDiscoveryInputMetaData svDiscoveryInputMetaData) {
+        final List<VariantContext> variantsWithFilterApplied = new ArrayList<>(variants.size());
+        final List<StructuralVariantFilter> filters = Arrays.asList(
+                new SVMQFilter(svDiscoveryInputMetaData.getDiscoverStageArgs().minMQ),
+                new SVAlnFilter(svDiscoveryInputMetaData.getDiscoverStageArgs().minAlignLength));
+        for (final VariantContext variant : variants) {
+            String svType = variant.getAttributeAsString(GATKSVVCFConstants.SVTYPE, "");
+            if (svType.equals(GATKSVVCFConstants.SYMB_ALT_ALLELE_DEL) || svType.equals(GATKSVVCFConstants.SYMB_ALT_ALLELE_INS) || svType.equals(GATKSVVCFConstants.SYMB_ALT_ALLELE_DUP)) {
+                if (Math.abs(variant.getAttributeAsInt(GATKSVVCFConstants.SVLEN, 0)) < StructuralVariationDiscoveryArgumentCollection.STRUCTURAL_VARIANT_SIZE_LOWER_BOUND )
+                    continue;
+            }
+            variantsWithFilterApplied.add(postHocFilterVariant(variant, filters));
+        }
+
+        final String out = outputPrefixWithSampleName + "merged_simple.vcf";
```

Comment:
I'd make a constant somewhere for the name `merged_simple.vcf`.

Reply:
done